rest-server/core/src/main/scala/com/actionml/core/engine/Engine.scala

```diff
@@ -30,9 +30,8 @@ import com.typesafe.scalalogging.LazyLogging
   */
 abstract class Engine extends LazyLogging with JsonParser {
 
-  // Todo: not sure how to require a val dataset: Dataset, which takes a type of Event parameter Dataset[CBEvent]
-  // for instance. Because each Dataset may have a different parameter type
-  var engineId: String = _
+  lazy val engineId: String = eid
+  private var eid: String = _
   private var mirroring: Mirroring = _
```

Comment:
@andreyshikov This gets rid of the public var but still requires delayed initialization.